### PR TITLE
Add WebGPU target doc

### LINKED ISSUE #5698: User guide: Create a 'targets' page for WebGPU
Create a similar 'targets' page for WebGPU/WGSL as for the other targets.
The one for Metal is probably a reasonable template.

https://shader-slang.com/slang/user-guide/targets.html

### PR DESCRIPTION
**Pull request by aleino-nv**
This closes #5698.

### CODE REVIEW

File: docs/user-guide/09-targets.md [hunk 1]
```diff
@@ -412,6 +412,45 @@ Slang's CPU compute target supports only a compute pipeline.
...
 Because CPU target support flexible pointer-based addressing and large low-latency caches, a compute kernel can simply be passed a small fixed number of pointers and be relied upon to load parameter values of any types via indirection through those pointers.
 
+## WebGPU
+
+> #### Note
+>
+> Slang support for WebGPU is work in progress.
+
+WebGPU is a graphics and compute API.
+It is similar in spirit to modern API's, like Metal, Direct3D12 and Vulkan, but with concessions to portability and privacy.
```
Comment (expipiplus1) [inline docs/user-guide/09-targets.md:422]: APIs

Comment (expipiplus1) [reply docs/user-guide/09-targets.md:422]: Direct3D 12

File: docs/user-guide/09-targets.md [hunk 2]
```diff
@@ -412,6 +412,45 @@ Slang's CPU compute target supports only a compute pipeline.
...
+
+### Parameter Passing
+
+WebGPU uses groups of bindings called bind groups to bind things like textures, buffers and samplers.
+Bind group objects are passed as arguments when encoding bind group setting commands.
+
+There is a notion of equivalence for bind groups, and a notion of equivalence for pipelines defined in
+terms of bind group equivalence.
+This equivalence allows an application to save some bind group setting commands, when switching between
+pipelines, if bindings are grouped together appropriately.
```
Comment (expipiplus1) [inline docs/user-guide/09-targets.md:452]: Might be nice to relate this to slang

**Review by expipiplus1 — CHANGES_REQUESTED** [review]

File: docs/user-guide/09-targets.md (hunk 1 above)
Comment (aleino-nv) [reply docs/user-guide/09-targets.md:422]: Done.

File: docs/user-guide/09-targets.md (hunk 2 above)
Comment (aleino-nv) [reply docs/user-guide/09-targets.md:452]: I added a mention of `ParamterBlock`. IIUC that is how you would control what bindings get grouped together, on the Slang side. (?)

However, it seems most of the other corresponding sections don't tie things back to Slang.

**Review by expipiplus1 — APPROVED** [review]

**Pull request merged**